File: targets/java-web/src/main/java/com/example/vulnapp/vulns/Vuln18PredictableToken.java
package com.example.vulnapp.vulns;

import com.example.vulnapp.vulns.crypto.Secrets;
import org.springframework.web.bind.annotation.*;

@RestController
public class Vuln18PredictableToken {
    @GetMapping("/vuln18/token")
    public String token(@RequestParam String user) {
        // predictable-token sink lives in Secrets.weakToken (VULN-18); seeded from username
        return Secrets.weakToken(user.hashCode());
    }
}

File: targets/kotlin-web/src/main/kotlin/com/example/vulnapp/vulns/Vuln18PredictableToken.kt
package com.example.vulnapp.vulns

import org.springframework.web.bind.annotation.*


@RestController
class Vuln18PredictableToken {

    @GetMapping("/vuln18/token")
    fun token(@RequestParam user: String): String = com.example.vulnapp.vulns.crypto.Secrets.weakToken(user.hashCode().toLong())
}
